File: exercicios/dia_016_strings_5/group_anagrams.cpp
class Solution {
public:
	vector<vector<string>> groupAnagrams(vector<string>& strs) {
		map<vector<int>, vector<string>> anagramsMap;

		for (string str : strs) {
			vector<int> charFreq(26);
			for (int i = 0; i < str.size(); i++) {
				charFreq[str[i] - 'a']++;
			}
			anagramsMap[charFreq].push_back(str);
		}

		vector<vector<string>> anagramsGroup;
		for (auto group : anagramsMap) {
			anagramsGroup.push_back(group.second);
		}

		return anagramsGroup;
	}
};

File: exercicios/dia_016_strings_5/group_anagrams.py
from collections import defaultdict

class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        hashmap = defaultdict(list)
        for string in strs:
            char_freq = [0]*26
            for c in string:
                char_freq[ord(c)-ord('a')] += 1
            string_freq = '|'.join([str(freq) for freq in char_freq])
            hashmap[string_freq].append(string)
        
        return hashmap.values()